File: spark/left-semi-join/relevant_users.scala
pupil.join(innovation_craft.where(col("type") === 1),
           (pupil.col("id") === innovation_craft.col("owner_id")),
           "left_semi")
     .select("id", "first_name", "last_name", "grade")
     .orderBy("grade", "id")
     .show()

File: spark/left-semi-join/relevant_users.py
pupil.join(innovation_craft.where(col("type") == 1),
           pupil["id"] == innovation_craft["owner_id"],
           "left_semi") \
     .select("id", "first_name", "last_name", "grade") \
     .orderBy("grade", "id") \
     .show()